File: tests/drivers/node/v5_8/write_routing.js
const neo4j = require('neo4j-driver');

function die() {
  session.close();
  driver.close();
  process.exit(1);
}

function Neo4jService(uri) {
  const driver = neo4j.driver(uri, neo4j.auth.basic("", ""));

  async function createGreeting() {
    const session = driver.session({ defaultAccessMode: neo4j.session.WRITE });
    try {
      const result = await session.writeTransaction(tx =>
        tx.run('CREATE (n:Greeting {message: "Hello NodeJs"}) RETURN n.message AS message')
      );
      console.log("Write txn finished");
    } finally {
      await session.close();
    }
  }

  async function close() {
    await driver.close();
  }

  return {
    createGreeting,
    close
  };
}

async function createGreetingsFromUri(uri) {
  const service = Neo4jService(uri);
  await service.createGreeting();
  await service.close();
}

async function main() {
  console.log("Started writing route");
  const uris = [
    'neo4j://localhost:7690',
    'neo4j://localhost:7691',
    'neo4j://localhost:7692'
  ];

  try {
    for (const uri of uris) {
      await createGreetingsFromUri(uri);
    }
  } catch (error) {
    console.error('An error occurred:', error);
    die();
  }
  console.log("Finished writing route");
}

main().catch(error => console.error(error));

File: tests/drivers/python/v5_8/write_routing.py
from neo4j import GraphDatabase


class Neo4jService:
    def __init__(self, uri, user="", password=""):
        self.driver = GraphDatabase.driver(uri, auth=(user, password))

    def close(self):
        self.driver.close()

    def create_greeting(self):
        with self.driver.session() as session:
            session.execute_write(self._create_and_return_greeting)
            print("Write txn passed!")

    @staticmethod
    def _create_and_return_greeting(tx):
        tx.run("CREATE (n:Greeting {message: 'Hello from Python'}) RETURN n.message AS message")


def create_greetings_from_uri(uri):
    service = Neo4jService(uri)
    service.create_greeting()
    service.close()


def main():
    print("Started writing route")
    uris = ["neo4j://localhost:7690", "neo4j://localhost:7691", "neo4j://localhost:7692"]

    try:
        for uri in uris:
            create_greetings_from_uri(uri)
    except Exception as error:
        print(f"An error occurred: {error}")
        exit(-1)
    print("Finished writing route")


if __name__ == "__main__":
    main()
